### DISCUSSION TITLE
Bleak support?

### DISCUSSION THREAD

**Discussion by 7wells** [micropython/micropython#13215] [Other] [answered]
Hello!

First, please accept my apologies if I placed my question in the wrong sub-forum. I'm not sure where it belongs and appreciated if moderators moved it to the right place, if necessary. Thank you! 😊

Now to my question:

Is there a way that **Micro**Python supports `bleak`, a cross platform Bluetooth Low Energy Client for Python using asyncio?

In other words:
Is it possible to port `bleak` from Python to MicroPython? (for e.g. ESP32)

For `bleak`, see there:
https://github.com/hbldh/bleak

Thanks for your hints! 👍

**Comment by andrewleech** [comment, marked as answer]
There's no direct support for bleak or its exact API as such.
Micropython has aioble for asyncio Bluetooth: https://github.com/micropython/micropython-lib/tree/master/micropython/bluetooth/aioble

**Reply by AxelPanningDE** [reply]
According to this page https://docs.micropython.org/en/latest/library/index.html asyncio is "micro-ified". I dont see there when this happened. Just looked it up right now. Your answer is 10 months back. 
Would that be good enough?

**Comment by fl4p** [comment]
Hi, I've started writing a wrapper so you can use the bleak API in micropython.
https://github.com/fl4p/micropython-bleak
